allow private repo

ryofujimura opened the pull request:
Allows user to display their private repository code.
Needs:
Github Private access token
User can obtain private access token under https://github.com/settings/tokens.

ryofujimura committed:
ui, option to add private repo and personal access token

ryofujimura committed:
github private repo code

ryofujimura committed:
fix public requiring token

no token for public.
token needed for private.

nuotsu commented:
@ryofujimura i thiiink there might be a *minor* security flaw here since it exposes the personal access token in the iframe src URL:

<img width="728" alt="Screenshot 2025-03-13 at 6 27 48 PM" src="https://github.com/user-attachments/assets/d965e137-0639-4cca-aa9b-f70735d3aea2" />

Also, for future commits, let's use Prettier formatter to keep code preferences consistent ([`.prettierrc`](https://github.com/nuotsu/github-iframe/blob/main/.prettierrc)).

ryofujimura committed:
Update domain to github-iframe-kappa.vercel.app

Replace all occurrences of github-iframe.vercel.app with github-iframe-kappa.vercel.app in store.ts and README.md

ryofujimura committed:
Update domain to github-iframe-kappa.vercel.app

- Update DOMAIN constant in store.ts to new Vercel deployment
- Update all example URLs in README.md
- All references now point to github-iframe-kappa.vercel.app

ryofujimura committed:
Fix private repo authentication

- Change Authorization header from 'token' to 'Bearer' format (correct GitHub API format)
- Use token immediately when provided (optimized for private repos)
- Improve error handling with clearer error messages
- Only fallback to raw URL for public repos (private repos can't use raw URLs)

ryofujimura committed:
Fix 404 errors by detecting and using default branch

- Add function to dynamically detect repository default branch
- Retry API call with explicit default branch if 404 occurs
- Use correct default branch in GitHub links (not hardcoded 'main')
- Improve 404 error messages with branch information
- Optimize branch detection to avoid duplicate API calls

ryofujimura committed:
Revert "Fix 404 errors by detecting and using default branch"

This reverts commit 4942faedb9796762398ffbc67c61293f380e4af2.

ryofujimura committed:
Revert "Fix private repo authentication"

This reverts commit 40d8fe0e62b8dc80308db267acc0dbfd3d458dfb.

nuotsu commented:
@ryofujimura Build error occurs
<img width="1271" height="812" alt="Screenshot 2025-11-05 at 11 07 51 AM" src="https://github.com/user-attachments/assets/8ef335ec-3935-40a4-8634-b5594be68e5b" />

```sh
11:06:14.348
    ▲ Next.js 15.1.6
11:06:14.349
 
11:06:14.373
    Creating an optimized production build ...
11:06:26.916
  ✓ Compiled successfully
11:06:26.921
    Linting and checking validity of types ...
11:06:31.077
    Collecting page data ...
11:06:33.913
    Generating static pages (0/4) ...
11:06:34.768
    Generating static pages (1/4) 
11:06:34.769
    Generating static pages (2/4) 
11:06:34.769
    Generating static pages (3/4) 
11:06:34.769
 Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
11:06:34.769
 HttpError: Bad credentials - https://docs.github.com/rest
11:06:34.769
     at A (/vercel/path0/.next/server/app/(site)/page.js:17:36864)
11:06:34.769
     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
11:06:34.769
     at async eb (/vercel/path0/.next/server/app/(site)/page.js:18:83459)
11:06:34.769
     at async u.doExecute (/vercel/path0/.next/server/app/(site)/page.js:16:7931)
11:06:34.769
 Export encountered an error on /(site)/page: /, exiting the build.
11:06:34.773
  ⨯ Static worker exited with code: 1 and signal: null
11:06:34.809
 Error: Command "npm run build" exited with 1
```
